src/ducks/classifications.js

```diff
@@ -121,10 +123,83 @@ const createClassification = () => {
   };
 };
 
+const saveAllQueuedClassifications = (dispatch) => {
+  const queue = JSON.parse(localStorage.getItem(FAILED_CLASSIFICATION_QUEUE_NAME));
+  let shownAlert = false;
+
+  if (queue && queue.length !== 0) {
+    console.log('Saving queued classifications:', queue.length);
+    const newQueue = [];
+    queue.forEach((classificationData) => {
+      apiClient.type('classifications').create(classificationData).save()
+        .then((actualClassification) => {
+          console.log('Saved classification', actualClassification.id);
+
+          try {
+            Split.classificationCreated(actualClassification);
+          } catch (err) { console.error('Split.classificationCreated() error: ', err); }
+
+          actualClassification.destroy();
+
+          try {
+            localStorage.setItem(FAILED_CLASSIFICATION_QUEUE_NAME, JSON.stringify(queue));
```

Comment:
Why is the current queue being saved to local storage here?

Reply:
I'm not sure if this should be `newQueue` or if you should delete this line completely.

Reply:
Thanks for spotting this; this is incorrect. This line of code was originally used to... shrink the queue at each step, I think, and should be changed to something like...

```
const queue = JSON.parse(localStorage.getItem(FAILED_CLASSIFICATION_QUEUE_NAME));
localStorage.setItem(FAILED_CLASSIFICATION_QUEUE_NAME, /* null or '' or '[]' */ );  //Reset queue, then the queue will be refilled later on failed submissions.  
```